h3/src/client.rs

```diff
@@ -209,17 +210,40 @@ where
     }
 }
 
-pub struct RequestStream<S> {
+pub struct RequestStream<T>(RequestStreamImpl<FrameStream<T>>)
+where
+    T: quic::RecvStream;
+
+impl<T> Deref for RequestStream<T>
+where
+    T: quic::RecvStream,
+{
+    type Target = RequestStreamImpl<FrameStream<T>>;
+    fn deref(&self) -> &Self::Target {
+        &self.0
+    }
+}
+
+impl<T> DerefMut for RequestStream<T>
+where
+    T: quic::RecvStream,
+{
+    fn deref_mut(&mut self) -> &mut Self::Target {
+        &mut self.0
+    }
+}
+
+pub struct RequestStreamImpl<S> {
```

Comment:
Instead of making a wrapper type, does the generic need to contain `FrameStream<T>` at all? Is there ever a use where a different generic is used internally? If not, you might be able to just have it be:

```rust
pub struct RequestStream<T> {
    inner: connection::RequestStream<FrameStream<T>, Bytes>,
}
```

Reply:
Why didn't I even think of that !